### PR TITLE
Add euler to angle axis conversion.

### PR DESCRIPTION
**Pull request by fvantilburg**
Implemented conversion of multiple Euler angle conventions to Angle Axis following: https://ntrs.nasa.gov/api/citations/19770024290/downloads/19770024290.pdf

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -41,6 +41,43 @@ fn convert_axis_angle_to_euler_xyz(rx: f64, ry: f64, rz: f64, angle: f64) -> (f6
...
+///
+/// * `euler_rx` - The rotation around the X axis.
+/// * `euler_ry` - The rotation around the Y axis.
+/// * `euler_rz` - The rotation around the Z axis.
+///
+/// Known robots that use this format:
+/// - Staubli (NOTE: this example assumes radians values, but this robot uses degrees)
+fn convert_euler_xyz_to_axis_angle(euler_rx: f64, euler_ry: f64, euler_rz: f64) -> (f64, f64, f64, f64) {
+
+	let cx = cos(euler_rx/2.0);
```
Comment (hgaiser) [inline src/main.rs]:
```suggestion
	let cx = cos(euler_rx / 2.0);
```

Let's stick to the normal formatting for operations.

File: src/main.rs [hunk 2]
```diff
@@ -41,6 +41,43 @@ fn convert_axis_angle_to_euler_xyz(rx: f64, ry: f64, rz: f64, angle: f64) -> (f6
...
+/// # Arguments
+///
+/// * `euler_rx` - The rotation around the X axis.
+/// * `euler_ry` - The rotation around the Y axis.
+/// * `euler_rz` - The rotation around the Z axis.
+///
+/// Known robots that use this format:
+/// - Staubli (NOTE: this example assumes radians values, but this robot uses degrees)
+fn convert_euler_xyz_to_axis_angle(euler_rx: f64, euler_ry: f64, euler_rz: f64) -> (f64, f64, f64, f64) {
+
```
Comment (hgaiser) [inline src/main.rs]:
```suggestion
```

File: src/main.rs [hunk 3]
```diff
@@ -41,6 +41,43 @@ fn convert_axis_angle_to_euler_xyz(rx: f64, ry: f64, rz: f64, angle: f64) -> (f6
...
+	let cx = cos(euler_rx/2.0);
+	let cy = cos(euler_ry/2.0);
+	let cz = cos(euler_rz/2.0);
+	let sx = sin(euler_rx/2.0);
+	let sy = sin(euler_ry/2.0);
+	let sz = sin(euler_rz/2.0);
+
+	// Compute the quaternion
+	let qw = -sx*sy*sz + cx*cy*cz;
```
Comment (hgaiser) [inline src/main.rs]:
```suggestion
	let qw = -sx * sy * sz + cx * cy * cz;
```

Similarly here. If you want to make the grouping clearer, we can add parenthesis around the two groups?

**Review by hgaiser — COMMENTED** [review]
Some minor styling issue, but overall looks good to me. Can you also add unit tests for these functions?

File: src/main.rs [hunk 4]
```diff
@@ -41,6 +41,43 @@ fn convert_axis_angle_to_euler_xyz(rx: f64, ry: f64, rz: f64, angle: f64) -> (f6
...
+///
+/// All rotations are expected and returned as radians.
+///
+/// # Arguments
+///
+/// * `euler_rx` - The rotation around the X axis.
+/// * `euler_ry` - The rotation around the Y axis.
+/// * `euler_rz` - The rotation around the Z axis.
+///
+/// Known robots that use this format:
```
Comment (vcarpani) [inline src/main.rs]:
```suggestion
/// Known robots that use Euler XYZ:
```
To specify; might be also good to change the rest of the comments to make it uniform

**Review by vcarpani — APPROVED** [review]

**Pull request merged**